server/src/api/schemas.py

```diff
@@ -72,6 +73,24 @@ class JobId(Schema):
     job_id = fields.String(required=True)
 
 
+class JobSearchRequest(Schema):
+    """Job search request schema"""
+
+    tags = fields.List(fields.String, description="List of tags to search for")
+    match = fields.String(
+        description="Match mode - 'all' or 'any' (default 'any')"
+    )
```

Comment:
We could enforce 'all' or 'any' via the `OneOf` [data validator](https://apiflask.com/schema/#data-validators). Then there is no need for us to check it in the code

Reply:
We don't currently have this as an enum or anything, and I could see it being soething that we change a bit soon. So it's only strings at the moment. Also, if we do end up wanting to add some "fake" ones that encompass more than one state it could affect this (see above)

Reply:
we're also checking that "tag" exists at all, see the comments below.

Reply:
Oh definitely, then something to consider once it's an enum